File: validation/src/main/java/com/validation/configuration/customValidate/CapitalizedConstraint.java
package com.validation.configuration.customValidate;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

@Documented
@Constraint(validatedBy = CapitalizedValidator.class)
@Target( { ElementType.METHOD, ElementType.FIELD })
@Retention(RetentionPolicy.RUNTIME)
public @interface CapitalizedConstraint {
    String message() default "in hoa!";
    Class<?>[] groups() default {};
    Class<? extends Payload>[] payload() default {};
}

File: spring-validate/src/main/kotlin/com/springvalidate/config/customValidate/CapitalizedConstraint.kt
package com.springvalidate.config.customValidate

import org.springframework.util.StringUtils
import java.lang.annotation.Documented
import javax.validation.Constraint
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext
import kotlin.reflect.KClass


@Documented
@Constraint(validatedBy = arrayOf(CapitalizedValidator::class))
@Target(*[AnnotationTarget.FIELD,AnnotationTarget.FUNCTION])
@Retention(AnnotationRetention.RUNTIME)
annotation class CapitalizedConstraint(
    val message: String = "Chữ đầu tiên phải được in hoa!",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<Any>> = []
)

class CapitalizedValidator : ConstraintValidator<CapitalizedConstraint?, String?> {
//    override fun initialize(constraintAnnotation: CapitalizedConstraint?) {
//        ConstraintValidator.su.initialize(constraintAnnotation)
//    }

    override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean {
        if (!StringUtils.hasLength(value)) return false
        if (!Character.isUpperCase(value!!.get(0))) return false
        return true
    }
}